arcgis-ios-sdk-samples/Analysis/Statistical query (group and sort)/StatisticalQueryGroupAndSortViewController.swift

```diff
@@ -28,9 +28,23 @@ class StatisticalQueryGroupAndSortViewController: UIViewController, UITableViewD
     private var orderByFields = [AGSOrderBy]()
     private var selectedOrderByFields = [AGSOrderBy]()
     private var statisticDefinitions = [AGSStatisticDefinition]()
-    private var sectionHeaderTitles = ["1. Add Statistic Definitions", "2. Select Group By Fields", "3. Select Order By Fields"]
     private var statisticTypes = ["Average", "Count", "Maximum", "Minimum", "StandardDeviation", "Sum", "Variance"]
     
+    private enum Section: Int, CaseIterable {
```

Comment:
Since you're conforming to `CaseIterable`, you could eliminate the raw value and avoid the optional initializer by using `allCases` to do the lookup:

```swift
switch Section.allCases[section] { … }
```

Please do that instead.

Reply:
I see your point but this seems less standard to me than using the initializer and we don't use this pattern anywhere else in the sample. Plus it'll still crash if the index is out of bounds so it's not any safer than force-unwrapping the init. I'll still change it if you really want.

Reply:
> this seems less standard to me than using the initializer and we don't use this pattern anywhere else in the sample

`CaseIterable` is a fairly new protocol. Still, we wouldn't be the first ones to [use that approach](https://oleb.net/blog/2018/06/enumerating-enum-cases/).

> Plus it'll still crash if the index is out of bounds so it's not any safer than force-unwrapping the init

It isn't any safer, but it isn't any less safe. The advantage is 1) not needing a raw representation for the enum cases, and 2) not needing to deal with optionals. 

> I'll still change it if you really want.

I would. We're already taking advantage of the `CaseIterable` conformance in `numberOfSections(in:)`. It would be nice to take advantage of it in the other data source methods and get rid of the raw values.